Modules/socketmodule.c

```diff
@@ -1719,18 +1719,32 @@ getsockaddrarg(PySocketSockObject *s, PyObject *args,
     case AF_UNIX:
     {
         Py_buffer path;
+        PyObject *fspath;
         int retval = 0;
 
-        /* PEP 383.  Not using PyUnicode_FSConverter since we need to
-           allow embedded nulls on Linux. */
-        if (PyUnicode_Check(args)) {
-            if ((args = PyUnicode_EncodeFSDefault(args)) == NULL)
-                return 0;
-        }
-        else
+        fspath = PyOS_FSPath(args);
+        if (fspath) {
+            args = fspath;
+            /* PEP 383.  Not using PyUnicode_FSConverter since we need to
+               allow embedded nulls on Linux. */
+            if (PyUnicode_Check(args)) {
+                PyObject *encoded_path = PyUnicode_EncodeFSDefault(args);
+                Py_DECREF(args);
+                args = encoded_path;
+                if (args == NULL)
+                    return retval;
+            }
+        } else {
+            PyErr_Clear();
             Py_INCREF(args);
+        }
+
         if (!PyArg_Parse(args, "y*", &path)) {
```

Comment:
You can simply use `PyObject_GetBuffer()` here (and maybe `PyObject_CheckBuffer()` if you want to look before you leap).

Reply:
`"y*"` also checks that the buffer is contiguous right? Do you think it's better to do `PyObject_GetBuffer()` and `PyBuffer_IsContiguous()`?

Reply:
If you use `PyBUF_SIMPLE`, `PyBuffer_IsContiguous()` is redundant.